server/verify.go

```diff
@@ -177,8 +189,8 @@ func getInstanceInfo(extensions []pkix.Extension) (*pb.GCEInstanceInfo, error) {
 	}, nil
 }
 
-// Checks if the provided AK public key can be trusted
-func validateAK(ak crypto.PublicKey, akCertBytes []byte, opts VerifyOpts) (*pb.MachineState, error) {
+// Verify the cert or key and return a MachineState.
+func getMachineStateFromCertOrKey(ak crypto.PublicKey, akCert *x509.Certificate, opts VerifyOpts) (*pb.MachineState, error) {
```

Comment:
pretty nit: but this function seems to be doing too much. Maybe we should pull the extraction of `MachineState` from *only the cert* into its own func? And then still call it `validateAK`

Reply:
Fixed in https://github.com/google/go-tpm-tools/pull/209/commits/352f92b26aa155950fd4b134e4091401ebe6afad to just have `validateAKPub` and `validateAKCert` functions. They are then conditionally called in `VerifyAttestation`